lib/matplotlib/legend_handler.py

```diff
@@ -148,21 +148,24 @@ def get_numpoints(self, legend):
     def get_xdata(self, legend, xdescent, ydescent, width, height, fontsize):
         numpoints = self.get_numpoints(legend)
 
-        if numpoints > 1:
+        if ((numpoints < 1) or not (numpoints.is_integer())):
```

Comment:
You should check for integralness first, I think. Also, most of these parentheses are unnecessary.